### DISCUSSION TITLE
An algorithm to suggest similar maps

### DISCUSSION THREAD

**Discussion by SevaOnOss** [ppy/osu#30301] [Ideas]
Currently it's very hard to find similar maps if you play a map you like. A simple algorithm like shopping sites' "People also bought this" would be very nice to see. Just look at what common maps are played in the same session between the people that also played the map you're looking at. I think the current API would be capable of doing this, and I don't think it would be that computationally expensive to do at all. And then just put a tab somewhere on the map listing with recommended maps based on the algorithm.
I think this would enhance the gameplay experience greatly, since I can just take the map I enjoyed playing and see similar maps that I'd also enjoy.

**Comment by Tenexxt** [comment]
This could be pulled of with the new beatmap tag system. Since every map would be tagged for its playstyle (tech, aim, speed etc.), The game could show you the maps with such tag, and genre tags already technically exist (you can mark the map with a genre but for the map to be detected for it it also has to be in the 'normal' tag section) you can just show the map with similar genres to ones you play lately.
Combine the two and you get the recommendation system. There could be some faults to this if I probably looked for it, but it should be possible to execute without much problems
